src/converters.js

```diff
@@ -447,7 +447,7 @@ export function cleanListItem( evt, data, conversionApi ) {
 
 		for ( const child of children ) {
 			if ( foundList && !child.is( 'ul' ) && !child.is( 'ol' ) ) {
-				child.remove();
+				child._remove();
```

Comment:
Same as above - use writer instead of protected methods which should not be used outside engine.